pkg/tsdb/elasticsearch/aggregation_factory.go

```diff
@@ -0,0 +1,173 @@
+			a.Size = size
+		} else {
+			a.Size = stringToIntWithDefaultValue(bucketAgg.Settings.Get("size").MustString(), defaultSize)
+		}
+
+		if minDocCount, err := bucketAgg.Settings.Get("min_doc_count").Int(); err == nil {
+			a.MinDocCount = &minDocCount
+		}
+		if missing, err := bucketAgg.Settings.Get("missing").String(); err == nil {
+			a.Missing = &missing
+		}
+
+		if orderBy, err := bucketAgg.Settings.Get("orderBy").String(); err == nil {
+			/*
+			   The format for extended stats and percentiles is {metricId}[bucket_path]
+			   for everything else it's just {metricId}, _count, _term, or _key
+			*/
+			metricIdRegex := regexp.MustCompile(`^(\d+)`)
+			metricId := metricIdRegex.FindString(orderBy)
+
+			if len(metricId) > 0 {
+				for _, m := range metrics {
+					if m.ID == metricId {
+						if m.Type == "count" {
+							a.Order["_count"] = bucketAgg.Settings.Get("order").MustString("desc")
+						} else {
+							a.Order[orderBy] = bucketAgg.Settings.Get("order").MustString("desc")
+							b.Metric(m.ID, m.Type, m.Field, nil)
+						}
+						break
+					}
+				}
+			} else {
+				a.Order[orderBy] = bucketAgg.Settings.Get("order").MustString("desc")
+			}
+		}
+
+		aggBuilder = b
+	})
+
+	return aggBuilder
+}
+
+// addNestedAgg adds a nested aggregation to the aggregation builder
+func addNestedAgg(aggBuilder es.AggBuilder, bucketAgg *BucketAgg) es.AggBuilder {
+	aggBuilder.Nested(bucketAgg.ID, bucketAgg.Field, func(a *es.NestedAggregation, b es.AggBuilder) {
+		aggBuilder = b
+	})
+
+	return aggBuilder
+}
+
+// addFiltersAgg adds a filters aggregation to the aggregation builder
+func addFiltersAgg(aggBuilder es.AggBuilder, bucketAgg *BucketAgg) es.AggBuilder {
+	filters := make(map[string]any)
+	for _, filter := range bucketAgg.Settings.Get("filters").MustArray() {
+		json := simplejson.NewFromAny(filter)
+		query := json.Get("query").MustString()
+		label := json.Get("label").MustString()
+		if label == "" {
+			label = query
+		}
+		filters[label] = &es.QueryStringFilter{Query: query, AnalyzeWildcard: true}
+	}
+
+	if len(filters) > 0 {
+		aggBuilder.Filters(bucketAgg.ID, func(a *es.FiltersAggregation, b es.AggBuilder) {
+			a.Filters = filters
+			aggBuilder = b
+		})
+	}
+
+	return aggBuilder
+}
+
+// addGeoHashGridAgg adds a geohash grid aggregation to the aggregation builder
+func addGeoHashGridAgg(aggBuilder es.AggBuilder, bucketAgg *BucketAgg) es.AggBuilder {
+	aggBuilder.GeoHashGrid(bucketAgg.ID, bucketAgg.Field, func(a *es.GeoHashGridAggregation, b es.AggBuilder) {
+		a.Precision = stringToIntWithDefaultValue(bucketAgg.Settings.Get("precision").MustString(), es.DefaultGeoHashPrecision)
+		aggBuilder = b
+	})
+
+	return aggBuilder
+}
+
+// isCalendarInterval checks if the interval is a calendar interval
+func isCalendarInterval(interval string) bool {
+	calendarIntervals := []string{"1w", "1M", "1q", "1y"}
+	for _, ci := range calendarIntervals {
+		if interval == ci {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
```suggestion
// isCalendarInterval checks if the interval is a calendar interval
func isCalendarInterval(interval string) bool {
	calendarIntervals := map[string]struct{}{
		"1w": {}, "1M": {}, "1q": {}, "1y": {},
	}
	_, ok := calendarIntervals[interval]
	return ok
}

```

We can cheat a bit and do this instead!